src/sass/parse.js

```diff
@@ -4860,11 +4874,22 @@ function getCompoundSelector1() {
     else if (checkAttributeSelector(pos)) sequence.push(getAttributeSelector());
     else if (checkPseudo(pos)) sequence.push(getPseudo());
     else if (checkPlaceholder(pos)) sequence.push(getPlaceholder());
+    else break;
```

Comment:
I generally add a break to stop any chance of mysterious problematic loops in the `get....` functions. Happy to remove though!